tests/e2e-playwright/tests/conftest.py

```diff
@@ -3,28 +3,192 @@
+        default=False,
+        help="User name for logging into the deployment",
+    )
+    group.addoption(
+        "--user-name",
+        action="store",
+        type=str,
+        default=None,
+        help="User name for logging into the deployment",
+    )
+    group.addoption(
+        "--password",
+        action="store",
+        type=str,
+        default=None,
+        help="Password for logging into the deployment",
+    )
+    group.addoption(
+        "--product-billable",
+        action="store_true",
+        default=False,
+        help="Whether product is billable or not",
+    )
+    group.addoption(
+        "--service-test-id",
+        action="store",
+        type=str,
+        default=None,
+        help="Service test ID",
+    )
+    group.addoption(
+        "--service-key",
+        action="store",
+        type=str,
+        default=None,
+        help="Service Key",
+    )
+
+
+@pytest.fixture(autouse=True)
 def osparc_test_id_attribute(playwright):
     # Set a custom test id attribute
     playwright.selectors.set_test_id_attribute("osparc-test-id")
 
 
 @pytest.fixture
 def api_request_context(context: BrowserContext):
+    return context.request
+
+
+@pytest.fixture
+def product_url(request: pytest.FixtureRequest) -> AnyUrl:
+    if passed_product_url := request.config.getoption("--product-url"):
+        return TypeAdapter(AnyUrl).validate_python(passed_product_url)
+    return TypeAdapter(AnyUrl).validate_python(os.environ["PRODUCT_URL"])
+
+
+@pytest.fixture
+def user_name(request: pytest.FixtureRequest, auto_register: bool, faker: Faker) -> str:
+    if auto_register:
+        faker.seed_instance(random.randint(0, 10000000000))  # noqa: S311
+        return faker.email()
+    if osparc_user_name := request.config.getoption("--user-name"):
+        assert isinstance(osparc_user_name, str)
+        return osparc_user_name
+    return os.environ["USER_NAME"]
+
+
+@pytest.fixture
+def user_password(
+    request: pytest.FixtureRequest, auto_register: bool, faker: Faker
+) -> str:
+    if auto_register:
+        return faker.password(length=12)
+    if osparc_password := request.config.getoption("--password"):
+        assert isinstance(osparc_password, str)
+        return osparc_password
+    return os.environ["USER_PASSWORD"]
+
+
+@pytest.fixture
+def product_billable(request: pytest.FixtureRequest) -> bool:
+    billable = request.config.getoption("--product-billable")
+    return TypeAdapter(bool).validate_python(billable)
+
+
+@pytest.fixture
+def service_test_id(request: pytest.FixtureRequest) -> str:
+    if test_id := request.config.getoption("--service-test-id"):
+        assert isinstance(test_id, str)
+        return test_id
+    return os.environ["SERVICE_TEST_ID"]
+
+
+@pytest.fixture
+def service_key(request: pytest.FixtureRequest) -> str:
+    if key := request.config.getoption("--service-key"):
+        assert isinstance(key, str)
+        return key
+    return os.environ["SERVICE_KEY"]
```

Comment:
Are these documented/collected somewhere? If not it might make sense to collect them all in a pydantic model, at least to "document" them in a single place. Just a suggestion...

Reply:
I see what you did here. These are mostly optional. And we have an easy way to overwrite them via CLI.
I think adding a pedantic model to capture the setting sin one single place would have been perfect, but I do understand that it makes things more complicated.
This solutions works fine for me.  I would also add a note in the doctoring of the addoption above that there is a fallback to the env vars.

Reply:
ACtually I would totally suppress the environment variable in a second step.

Reply:
@bisgaard-itis : these are not documented anywhere. Basically directly used at the moment in e2e-backend. But my plan was to totally remove that soon.

Reply:
OK I prefer the suppression as well in the long run